Add broadcasts for most guild events

itsneyoa opened the pull request:
Added broadcasts for all the following guild events:
- Player logins and logouts
- Guild joins, leaves and kicks
- Guild rank promotions and demotions
- Hypixel error messages (`You cannot say the same message twice`, `We blocked your comment...`, `You do not have permission to use this command!` etc)
- Online and offline guild invite feedback
- User and guild chat mutes and unmutes
---
All the broadcasts are sent in the regular guild chat channel as that's the only place commands work and are roughly colour coded on if they're positive or negative. Command feedback built into the bot has also been removed, instead it relies on Hypixel's `Invalid usage!` feedback as shown below.  

![omg leaking guild chat D:](https://user-images.githubusercontent.com/71194294/121440211-622bcf80-c97f-11eb-947a-3ff42f142d0d.png)
![oh noe more top secret leaks](https://user-images.githubusercontent.com/71194294/121441212-445f6a00-c981-11eb-960e-eca0d8644d35.png)

Thanks to PR #22 for reminding me I'd started this :)

SaegusaMayumi1234 commented:
well should I close my pr? I think yours is better so #22

itsneyoa commented:
> It has different purpose but idk

As far as I can see the main differences are:
- Logging to a separate channel
- Being togglable in the configuration
- Formatting of the messages to include/exclude the person who executed the command

As the Discord commands will only work in the command specified in the configuration, in my opinion it makes more sense to log all events there - running the mute command only to have to switch channels to ensure it worked seems pointless.
Additionally some events are broadcasted to all guild members anyway, however events like mutes aren't so it may be a good idea to put them in a different channel. @SergeantSar is working on officer chat support at the moment, so that may tie nicely into this.

Being togglable in the configuration could be easily added, however I'm not sure if it's really necessary.

The formatting of the message wasn't something I gave much thought about, I didn't see any benefit from adding the executing staff member to the messages so decided to leave it out, however again if we were to add the events to an officer chat channel it may be better to show the staff member in order to identify and prevent staff abuse.

Again, this is all what I thought made the most sense so if you disagree or have any other suggestions please leave them here :)

SaegusaMayumi1234 commented:
Well I think so.

Anyways this is not really necessary but I just spotted missing error message like:
-You can only demote up to your own rank!
-They already have that rank!
-${name} is already the lowest rank you've created!
-${name} is the guild master so can't be demoted!
-${name} is the guild master so can't be promoted anymore!
-You are sending commands too fast! Please slow down. (this is important because we use /gc (message) and this is considered as command. So when spam happen this message could occur)

Well you don't need to add this if you think this is not necessary to add but just for suggestion

itsneyoa commented:
I've added all those errors too now, it seems since I added most of the event replies something's changed so the `----------------------------------------------------` before and after the actual message messes up `.startsWith()` and `.endsWith()`, that's also been fixed.

SaegusaMayumi1234 commented:
Alright seems cool to me!

Senither reviewed (approved):
Looks good, thanks for the PR! 👍 

@SaegusaMayumi1234 
Thanks for helping review the PR and catching the missing messages, I wouldn't have found them by myself^^
